tests/unit/conftest.py

```diff
@@ -43,5 +63,42 @@ def config_path(tmpdir_factory):
             file,
             sort_keys=False,
         )
+    # bad input_data.yaml
+    with open(f"{tmp_path}/bad_input_data.yaml", "w") as file:
+        yaml.dump(
+            {
+                "foo": str(input_dir1 / "non-existent-file1"),
+                "bar": str(input_dir2 / "non-existent-file2"),
+            },
+            file,
+            sort_keys=False,
+        )
+
+    # bad implementation
+    bad_step_implementation_dir = tmp_path.join("steps/foo/implementations/bar/").ensure(
+        dir=True
+    )
+    with open(f"{bad_step_implementation_dir}/metadata.yaml", "w") as file:
+        yaml.dump(
+            {
+                "step": "foo",  # not a supported step
+                "image": {
+                    "path": "some/path/to/container/directory",
+                    "filename": "bar",
+                },
+            },
+            file,
+            sort_keys=False,
+        )
 
     return str(tmp_path)
+
+
+@pytest.fixture(scope="session")
+def config(test_dir) -> Config:
+    """A good/known Config object"""
+    return Config(
+        f"{test_dir}/pipeline.yaml",
+        f"{test_dir}/input_data.yaml",
+        f"{test_dir}/environment.yaml",
+    )
```

Comment:
It seems like you need all these files? There should be a test for that or a test that what users have provided is everything they need to provide to plug and play. Not sure if that is possible to write a test for yet though.

Reply:
They technically only _need_ the first two, but those are the so-called "required options" of the tool so will immediately error if they don't pass it in. The enviroenment.yaml is optional and I do have a test that if they don't provide it we get the expected default computing_environment == "local"